core/lib/health_check/src/lib.rs

```diff
@@ -118,30 +123,59 @@ impl Default for AppHealthCheck {
 }
 
 impl AppHealthCheck {
-    pub fn new(slow_time_limit: Option<Duration>, hard_time_limit: Option<Duration>) -> Self {
-        const DEFAULT_SLOW_TIME_LIMIT: Duration = Duration::from_millis(500);
-        const DEFAULT_HARD_TIME_LIMIT: Duration = Duration::from_secs(3);
+    const DEFAULT_SLOW_TIME_LIMIT: Duration = Duration::from_millis(500);
+    const DEFAULT_HARD_TIME_LIMIT: Duration = Duration::from_secs(3);
 
-        let slow_time_limit = slow_time_limit.unwrap_or(DEFAULT_SLOW_TIME_LIMIT);
-        let hard_time_limit = hard_time_limit.unwrap_or(DEFAULT_HARD_TIME_LIMIT);
+    pub fn new(slow_time_limit: Option<Duration>, hard_time_limit: Option<Duration>) -> Self {
+        let slow_time_limit = slow_time_limit.unwrap_or(Self::DEFAULT_SLOW_TIME_LIMIT);
+        let hard_time_limit = hard_time_limit.unwrap_or(Self::DEFAULT_HARD_TIME_LIMIT);
         tracing::debug!("Created app health with time limits: slow={slow_time_limit:?}, hard={hard_time_limit:?}");
 
-        let config = AppHealthCheckConfig {
-            slow_time_limit: slow_time_limit.into(),
-            hard_time_limit: hard_time_limit.into(),
+        let inner = AppHealthCheckInner {
+            components: Vec::default(),
+            slow_time_limit,
+            hard_time_limit,
+        };
+        Self {
+            inner: Mutex::new(inner),
+        }
+    }
+
+    pub fn override_limits(
```

Comment:
Both this method and `expose_metrics` are somewhat clunky; I think it would be easier to refactor it once we fully switch to the framework and remove the old code.